velox/functions/sparksql/tests/StringToMapTest.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::functions::sparksql::test {
+using namespace facebook::velox::test;
+namespace {
+class StringToMapTest : public SparkFunctionBaseTest {
+ protected:
+  void testStringToMap(
+      const std::vector<StringView>& inputs,
+      const std::vector<std::pair<StringView, std::optional<StringView>>>&
+          expect) {
+    std::vector<VectorPtr> row;
+    row.emplace_back(makeFlatVector<StringView>({inputs[0]}));
+    std::string expr =
+        fmt::format("str_to_map(c0, '{}', '{}')", inputs[1], inputs[2]);
+    auto result = evaluate<MapVector>(expr, makeRowVector(row));
```

Comment:
Maybe makeRowVector(row) -> makeRowVector({makeFlatVector<StringView>({input})}) and remove row.